libs/nox/src/tensor.rs

```diff
@@ -173,14 +173,17 @@ pub trait NonScalarDim {}
 /// Represents constant dimensions, specified at compile-time.
 pub trait ConstDim {}
 
+/// Represents a scalar dimension, which is essentially dimensionless.
 pub type ScalarDim = ();
 impl TensorDim for ScalarDim {}
 impl TensorDim for nalgebra::Dyn {}
 impl NonScalarDim for nalgebra::Dyn {}
 impl<const N: usize> TensorDim for nalgebra::Const<N> {}
 impl<const N: usize> NonScalarDim for nalgebra::Const<N> {}
 
+/// Trait for dimensions compatible with XLA computation, defining shape information.
 pub trait XlaDim {
+    /// Returns an empty shape, representing a scalar with no dimensions.
```

Comment:
This isn't necessarily the case. The function returns the shape of the implementing type, which may or may not be a scalar.